### PR TITLE
Nam service

### PR DESCRIPTION
**Pull request by NguyenNam34**

### CODE REVIEW

File: app/src/main/java/com/fstyle/androidtrainning/screen/main/MainActivity.java [hunk 1]
```diff
@@ -102,13 +186,97 @@ public void onStart() {
...
+        if (fromUser) {
+            mServicePlayMusic.setMediaLength(progress);
+            mServicePlayMusic.seekMusic();
+        } else {
+            mHandler.removeCallbacks(RSetupUI);
+            mHandler.postDelayed(RSetupUI, 1000);
+        }
+    }
+
+    private Runnable RSetupUI = new Runnable() {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/main/MainActivity.java]: WARNING: Name 'RSetupUI' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/fstyle/androidtrainning/screen/splash/SplashActivity.java [hunk 2]
```diff
@@ -22,6 +23,7 @@
     public static final int REQUEST_PERMISSION_CODE = 101;
     public static final int TIME_HANDLER = 3000;
     public static final int SDK_DEFAULT = 23;
+    private String TAG = "chuong";
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/splash/SplashActivity.java]: WARNING: Name 'TAG' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        return START_STICKY;
+    }
+
+    @Override
+    public void onDestroy() {
+        mMediaPlayer.release();
+        super.onDestroy();
+    }
+
+    private void createMusic(int id) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: WARNING: 'id' hides a field.

File: app/src/main/java/com/fstyle/androidtrainning/model/Attr.java [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
+import android.os.Parcelable;
 import com.google.gson.annotations.Expose;
 import com.google.gson.annotations.SerializedName;
 
 /**
  * Created by Administrator on 12/12/17.
  */
 
-public class Attr {
+public class Attr implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/model/Attr.java:12]: WARNING: [ShortClassName] Avoid short class names like Attr
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/model/Date.java [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
+import android.os.Parcelable;
 import com.google.gson.annotations.Expose;
 import com.google.gson.annotations.SerializedName;
 
 /**
  * Created by Administrator on 12/12/17.
  */
 
-public class Date {
+public class Date implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/model/Date.java:12]: WARNING: [ShortClassName] Avoid short class names like Date
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/screen/splash/SplashActivity.java (hunk 2 above)
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/splash/SplashActivity.java]: WARNING: [SuspiciousConstantFieldName] The field name indicates a constant but its modifiers do not
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#SuspiciousConstantFieldName

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        try {
+            String path = mExternalData.getArrayListTrack().get(id).getTrackData();
+            mUri = Uri.parse(path);
+            mMediaPlayer = new MediaPlayer();
+            mMediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
+            mMediaPlayer.setDataSource(path);
+            mMediaPlayer.prepare();
+            MediaPlayer.create(getApplicationContext(), mUri);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/com/fstyle/androidtrainning/screen/main/MainActivity.java [hunk 7]
```diff
@@ -1,33 +1,50 @@
...
 import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
 import com.fstyle.androidtrainning.R;
 import com.fstyle.androidtrainning.application.MainApplication;
 import com.fstyle.androidtrainning.data.remote.service.config.LastFmApi;
+import com.fstyle.androidtrainning.model.Track;
 import com.fstyle.androidtrainning.screen.BaseActivity;
+import com.fstyle.androidtrainning.screen.main.mainfragments.mysong.subfragment.listsong
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/main/MainActivity.java]: ERROR: Unused import: com.fstyle.androidtrainning.screen.main.mainfragments.mysong.subfragment.listsong.ListSongAdapter. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
+package com.fstyle.androidtrainning.service;
+
+import android.app.Service;
+import android.content.Intent;
+import android.media.MediaPlayer;
+import android.net.Uri;
+import android.os.IBinder;
+import android.provider.MediaStore;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: Unused import: android.provider.MediaStore. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        return START_STICKY;
+    }
+
+    @Override
+    public void onDestroy() {
+        mMediaPlayer.release();
+        super.onDestroy();
+    }
+
+    public void playMusic(){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        mMediaPlayer.start();
+    }
+
+    public void pauseMusic(){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        if (mMediaPlayer.isPlaying()){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+            mediaLenght = mMediaPlayer.getCurrentPosition();
+            mMediaPlayer.pause();
+        }
+    }
+
+    public void forwardMusic(){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+        if (mMediaPlayer.isPlaying()){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+            mMediaPlayer.stop();
+            mMediaPlayer.release();
+            mMediaPlayer = MediaPlayer.create(this, mLenght.get(position));
+            mMediaPlayer.start();
+        }else {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: 'else' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+            mMediaPlayer = MediaPlayer.create(this, mLenght.get(position));
+            mMediaPlayer.start();
+        }
+    }
+
+    public void seekMusic(){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+        mMediaPlayer.seekTo(mediaLenght);
+    }
+
+    public void stopMusic(){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/main/MainActivity.java (hunk 7 above)
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/main/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.fstyle.androidtrainning.screen.main.mainfragments.mysong.subfragment.listsong.ListSongAdapter'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java (hunk 8 above)
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.provider.MediaStore'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/androidtrainning/screen/splash/SplashActivity.java [hunk 17]
```diff
@@ -7,6 +7,7 @@
 import android.os.Handler;
 import android.support.annotation.NonNull;
 import android.support.v4.app.ActivityCompat;
+import android.util.Log;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/splash/SplashActivity.java]: ERROR: Unused import: android.util.Log. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/splash/SplashActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.util.Log'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

**Comment by tranminhtri911** [comment]
Ok

**Pull request merged**